app/src/main/java/org/openobservatory/ooniprobe/activity/RunningActivity.java

```diff
@@ -45,38 +37,61 @@
     ProgressBar progress;
     @BindView(R.id.animation)
     LottieAnimationView animation;
+    private ArrayList<AbstractSuite> testSuites;
     private AbstractSuite testSuite;
     private boolean background;
     private Integer runtime;
 
-    public static Intent newIntent(AbstractActivity context, AbstractSuite testSuite) {
+    public static Intent newIntent(AbstractActivity context, ArrayList<AbstractSuite> testSuites) {
         if (ReachabilityManager.getNetworkType(context).equals(ReachabilityManager.NO_INTERNET)) {
             new MessageDialogFragment.Builder()
                     .withTitle(context.getString(R.string.Modal_Error))
                     .withMessage(context.getString(R.string.Modal_Error_NoInternet))
                     .build().show(context.getSupportFragmentManager(), null);
             return null;
-        } else
-            return new Intent(context, RunningActivity.class).putExtra(TEST, testSuite);
+        } else {
+            Bundle extra = new Bundle();
+            extra.putSerializable(TEST, testSuites);
+            return new Intent(context, RunningActivity.class).putExtra(TEST, extra);
+        }
     }
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        testSuite = (AbstractSuite) getIntent().getSerializableExtra(TEST);
-        runtime = testSuite.getRuntime(getPreferenceManager());
-        setTheme(testSuite.getThemeDark());
+        setTheme(R.style.Theme_MaterialComponents_NoActionBar_App);
         setContentView(R.layout.activity_running);
         ButterKnife.bind(this);
+        Bundle extra = getIntent().getBundleExtra(TEST);
+        testSuites = (ArrayList<AbstractSuite>) extra.getSerializable(TEST);
+        if (testSuites == null)
+            finish();
+        runtTest();
+    }
+
+    private void runtTest() {
+        //TODO cycle array
```

Comment:
Does this mean that we are still run a single test at a time?

Reply:
Old todo, it has been done. Removed